hudi-common/src/main/java/org/apache/hudi/common/table/timeline/versioning/v2/ActiveTimelineV2.java

```diff
@@ -522,8 +520,10 @@ protected void transitionStateToComplete(boolean shouldLock, HoodieInstant fromI
               "Could not rename " + fromInstantPath + " to " + toInstantPath);
         }
       } else {
+        // Ensures old state exists in timeline
         ValidationUtils.checkArgument(
-            metaClient.getStorage().exists(getInstantFileNamePath(fromInstantFileName)));
+            metaClient.getStorage().exists(getInstantFileNamePath(fromInstantFileName)),
+            "Metadata file " + getInstantFileNamePath(fromInstantFileName) + " does not exist when transition state.");
```

Comment:
It looks like only this line needs a fix.

Reply:
Revert other lines.

Reply:
> It looks like only this line needs a fix.

Revert other lines.